torch/_logging/_internal.py

```diff
@@ -0,0 +1,424 @@
+    # registration by the user (ie placing "torch._dynamo.output_graph" in the env var)
+    # these need to be tracked so their levels can be reset properly
+    child_log_qnames: Set[str] = field(default_factory=set)
+
+    # artifact names, populated by register_artifact
+    artifact_names: Set[str] = field(default_factory=set)
+
+    # artifacts which are not displayed unless explicitly named in the
+    # settings. Ex. output_code is NOT displayed even if the inductor
+    # log level is set to DEBUG. It must be explicitly named in the settings
+    off_by_default_artifact_names: Set[str] = field(default_factory=set)
+
+    def is_artifact(self, name):
+        return name in self.artifact_names
+
+    def is_log(self, alias):
+        return alias in self.log_alias_to_log_qname
+
+    # register a log with an alias
+    def register_log(self, alias, log_qname):
+        self.log_alias_to_log_qname[alias] = log_qname
+
+    # register an artifact name
+    def register_artifact_name(self, name, off_by_default):
+        self.artifact_names.add(name)
+
+        # if off by default, don't enable it
+        # when log_name's log_level is set to DEBUG
+        if off_by_default:
+            self.off_by_default_artifact_names.add(name)
+
+    # register the qualified name of an artifact log
+    # this is needed to know which logs need to be reset
+    # whenever the log_state is changed
+    def register_artifact_log(self, artifact_log_qname):
+        self.artifact_log_qnames.add(artifact_log_qname)
+
+    def register_child_log(self, log_qname):
+        self.child_log_qnames.add(log_qname)
+
+    def get_log_qnames(self):
+        return set(self.log_alias_to_log_qname.values())
+
+    def get_artifact_log_qnames(self):
+        return set(self.artifact_log_qnames)
+
+    def get_child_log_qnames(self):
+        return set(self.child_log_qnames)
+
+    def is_off_by_default(self, artifact_qname):
+        return artifact_qname in self.off_by_default_artifact_names
+
+
+@dataclass
+class LogState:
+    # qualified log names -> currently set log level
+    log_qname_to_level: Dict[str, str] = field(default_factory=dict)
+
+    # the set of currently enabled artifacts
+    artifact_names: Set[str] = field(default_factory=set)
+
+    def enable_artifact(self, artifact_name):
+        self.artifact_names.add(artifact_name)
+
+    def is_artifact_enabled(self, name):
+        return name in self.artifact_names
+
+    def enable_log(self, log_qname, log_level):
+        self.log_qname_to_level[log_qname] = log_level
+
+    def get_log_level_pairs(self):
+        return self.log_qname_to_level.items()
+
+    def clear(self):
+        self.log_qname_to_level.clear()
+        self.artifact_names.clear()
+
+
+log_registry = LogRegistry()
+log_state = LogState()
+
+
+# User API for setting log properties
+# ex. format set_logs(LOG_NAME=LEVEL, ARTIFACT_NAME=bool)
+# ex. set_logs(dynamo=logging.DEBUG, graph_code=True)
+def set_logs(
```

Comment:
nit: make this kwarg only

Reply:
I liked the autocomplete benefit, this is easy enough to change if it gets too large